tools/trace.py

```diff
@@ -299,6 +299,26 @@ def _generate_field_assign(self, idx):
                                         (idx, Probe.c_type[field_type], expr)
                 self._bail("unrecognized field type %s" % field_type)
 
+        def _generate_usdt_filter_read(self):
+            text = ""
+            if self.probe_type == "u":
+                    for arg, _ in Probe.aliases.items():
```

Comment:
Ideally we should do this. The issue here is for filters, unlike the return (printf) values, we don't actually parse them. We just take what's in between brackets in the CLI argument and put it in the `if` statement directly, with some direct find-and-replace on variable names (as you can see here). We would have to parse the filter expression in order to get type and expr as we do for printf values:(